src/ape/pytest/runners.py

```diff
@@ -36,7 +36,11 @@ def __init__(
 
     @property
     def _provider_context(self) -> ProviderContextManager:
-        return self.network_manager.parse_network_choice(self.config_wrapper.network)
```

Comment:
To play devil's advocate here, since you have the ConfigError already raising in that other place, then it doesn't necessarily also need to be here in case somebody needs to run on mainnet for whatever reason. They just have to explicitly call it, it cannot be the default

Reply:
I was imagining someone automating a shell command poorly such that it resulted in mainnet being passed in accidentally

Reply:
I'd want to block where you set mainnet as the default network in config, but if you explictly set it by accident that's kind of on you. I don't think that would be common anyways, and I'm about 50-50 if it would be intentional (and thus annoying to be unable to do it by any means), so I am a bit biased against raising for explicitly setting it to mainnet

Reply:
OK, I will change it to allow running tests on mainnet.